crates/nu-protocol/src/shell_error.rs

```diff
@@ -627,12 +627,14 @@ pub enum ShellError {
     /// This error is fairly generic. Refer to the specific error message for further details.
     #[error("Unsupported input")]
     #[diagnostic(code(nu::shell::unsupported_input))]
-    UnsupportedInput(
-        String,
-        String,
-        #[label("{0}")] Span, // call head (the name of the command itself)
-        #[label("input type: {1}")] Span,
-    ),
+    UnsupportedInput {
+        msg: String,
+        input: String,
+        #[label("{msg}")]
+        msg_span: Span,
+        #[label("input type: {input}")]
```

Comment:
OK this preexisting format string does not make sense right now:

We have a dominating amount of `Value originates from here` messages in `input` that would get labelled with `input type: value originates from here` and also stuff like `input: format!("input type: {:?}", val.get_type())` which would duplicate that.

This is part of the pain we should resolve with the endeavour of #10698 

For now here the attribute should be: 
```suggestion
        #[label("{input}")]
```

Reply:
Fair. I want to stick to minimal changes when making the PR to make them easy to review so any critique can be focused and things like this can be brought up that make sense to do.

I hope this kind of PR and review pushes nushell toward resolving #10698, and I think a review of the many `GenericError`s will be part of that other issue.